docs/concepts/workloads/controllers/statefulset.md

```diff
@@ -222,6 +252,15 @@ updates to its `.spec.template` will not be propagated to its Pods.
 In most cases you will not need to use a partition, but they are useful if you want to stage an
 update, roll out a canary, or perform a phased roll out.
 
+## Selectors and Labels
+
+Same as other controllers, a StatefulSet utilizes selectors (`spec.selector` field) to identify pods that it should manage. For a pod to be managed by a StatefulSet, the pod needs to have labels matching the StatefulSet's selectors. However, the StatefulSet does not keep track whether a pod is created by it or another controller. The pods will persist when the StatefulSet is deleted (except when the user performs a [cascading deletion](https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/)).
+
+In Kubernetes 1.8 and later, selector of a StatefulSet is made immutable after creation. The `.spec.selector.matchLabels` field must match `.spec.template.metadata.labels` field during creation of the StatefulSet, and cannot be changed during future updates. As the selector is now immutable, a nil `.spec.selector.matchLabels` field will not be defaulted to a non-nil `.spec.template.metadata.labels` value during creation. Instead, this is considered as user error because both fields do not match. In addition, `.metadata.labels` field will also not be defaulted to match the `.spec.template.metadata.labels` field. The user has to explicitly set the `.metadata.labels` field, but it can be different from `.spec.template.metadata.labels` value.
+
+### Overlapping Selectors
+It is possible for a pod to be created with labels matching selectors of multiple StatefulSets (or multiple controllers of different types such as both StatefulSet and ReplicaSet). Although k8s allows overlapping selectors to exist, they result in undefined behaviors. This is considered as user error, in which the user needs to manage deletion of the pod manually.
+
```

Comment:
Add an indication that in 1.8 the selector will no longer be defaulted.

Reply:
Sure!